src/multiply_module.f90

```diff
@@ -567,40 +675,56 @@ subroutine prefetch(this_part,ahalo,a_b_c,bmat,icall,&
     type(primary_set) :: prim
     type(cover_set) :: gcs
     integer :: mx_mpp
-    integer :: this_part,icall,n_cont,myid,ilen2
+    integer :: this_part,n_cont,myid,ilen2
     type(matrix), dimension(:) :: bmat
     type(matrix_halo) :: ahalo
     type(comms_data) :: a_b_c
     integer(integ), dimension(:)  :: bind_rem,bind
     integer :: lenb_rem, tag
+    logical, intent(in), optional :: do_nonb
+    integer, optional :: request(2)
     real(double), dimension(lenb_rem) :: b_rem
     real(double) :: b(:)
     ! Local variables
     integer :: ncover_yz,ind_part,iskip,ind_last
     integer :: inode,ipart,nnode
+    logical :: do_nonb_local
 
+    ! Set non-blocking receive flag
+    do_nonb_local = .false.
+    if (present(do_nonb)) do_nonb_local = do_nonb
```

Comment:
Could we remove the `do_nonb` argument, and just check for `if(present(request))` here? Then you wouldn't have to check for that later? Is there a benefit of being able to handle a scenario where `do_nonb = .false.` gets passed in?